### PR TITLE
Initialize judging page front-end.

### LINKED ISSUE #85: Judging initial pages
- [ ] Develop these 3 pages for desktop + mobile (check with sarah for mobile it should be pretty simple if not no changes from desktop)
- [ ] Be sure to use the secondary buttons for the Who are you? page 
- [ ] Create a Dropdown component and make it look like the design when its closed - don't worry about implementing what it looks like when the dropdown is open Sarah is still working on those designs 
- [ ] For the back button maybe pass a prop to the primary button to make it smaller 
Screenshot 2022-11-30 at 12 41 57 AM

Screenshot 2022-11-30 at 12 38 19 AM

### PR DESCRIPTION
**Pull request by surbhi-gulati**
Implement front-end for selection of person type & welcome judge/ hacker pages.

Fixes [#85](https://github.com/HackBeanpot/mono-repo/issues/85).

Changelist:

- Introduce `SecondaryDropdown` which is a select element that is styled similar to SecondaryEventButton but with less rounded edges
- Create the new judging tab route and develop a new page for it
- Fix routing throughout live-site by prepending `/` to links to make them relative
- Create a `JudgingSelection` component to declare a user as hacker or judge
- Create a `JudgingScheduleWelcome` component to welcome the given user type
- Create `JudgingSchedule` components for Judge & Hacker types and populate both initially with the `JudgingScheduleWelcome` logic

Tested: manual

Screenshots: todo

### CODE REVIEW

**Comment by surbhi-gulati** [comment]
@jzhzhang66 & @mike10911 ready for review, only issue is I couldn't figure out centering of the buttons though LOL. Any suggestions on what I'm missing here?

File: live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx
Comment (mike10911) [inline live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx]: To get the button centered I would change this to `display:flex` and use the justify-content property to center it!

File: live-site/components/judging-section/judging-selection/JudgingSelection.tsx
Comment (mike10911) [inline live-site/components/judging-section/judging-selection/JudgingSelection.tsx]: I think that this should be judging/judges & judging/hackers

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx
Comment (mike10911) [inline live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: change to ./ if you are meaning to go back to the selection page

**Review by mike10911 — CHANGES_REQUESTED** [review]

**Review by mike10911 — CHANGES_REQUESTED** [review]
Screen Shot 2022-12-15 at 8 03 02 PM


### Some style changes to match designs
- some space is needed between these buttons
- text needs to be white
- light green border needed
- padding on the left and right sides needed on the inside of the buttons

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx
Comment (surbhi-gulati) [reply live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: ./ means the current directory ../ means the parent of the current directory, so this actually works right due to the extra folder

Comment (mike10911) [reply live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: ohhh rightt lol

Comment (mike10911) [reply live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: I was taking me all the way back home when I tested it on my machine

File: live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx
Comment (surbhi-gulati) [reply live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx]: THIS WORKS THANK YOU MIKE

**Review by mike10911 — CHANGES_REQUESTED** [review]
Screen Shot 2022-12-15 at 8 08 04 PM

### Style Changes

- white text
- padding to the right of the down arrow
- white border needed
- down arrow should be a bit bigger

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx
Comment (surbhi-gulati) [reply live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: omg cool hf in pittsburgh :D

**Review by mike10911 — CHANGES_REQUESTED** [review]
Screen Shot 2022-12-15 at 8 20 10 PM

This should be slimmer to match design!

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
THANK YOU SURBHI FOR ALL THIS AMAZING WORK!!!

+1 to everything mike said about making the UI more similar to the design !!

Also, the "I am a hacker" and "I am a judge" buttons and the selection dropdown button are noticeably bigger on the design (kinda going off the padding thing mike mentioned) so could you please change that?

File: live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx
Comment (jzhzhang66) [inline live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx]: For these type of things I like to keep them a bit more general like StyledJudgingWelcomeHeader in case the actual text ends up changing! But that's just a nitpick - up to you!

File: live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx
Comment (jzhzhang66) [inline live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx:13]: This should be brown according to the design - change the color

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx
Comment (jzhzhang66) [inline live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: This needs to be brown colored text

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]

**Comment by jzhzhang66** [comment]
Plz fix merge conflicts too!

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Ok I think for next steps, lets: 
- Make the changes Mike requested here 
Screenshot 2023-01-01 at 5 47 28 PM
Screenshot 2023-01-01 at 5 50 51 PM
- After you've merged your code on main into this one, get rid of the secondary dropdown component like you were mentioning before and use the styled component you created
- Could you please move the styled component out of typography though? It would make more sense for you to put it inside of a file in a styled-components folder inside of live-site (kind of like this one in shared-ui) 
Screenshot 2023-01-01 at 5 49 03 PM

**Review by mike10911 — CHANGES_REQUESTED** [review]
This is great!!! Just a couple of minor comments

- the back button is still taking me back to the homepage
- green border around the selection buttons needs to be implemented
- Selection buttons could use a little more letter spacing to reflect the design
- The arrow should be bigger and have some padding on the right to match the design below
Screen Shot 2023-01-05 at 6 00 39 PM

File: shared-ui/components/primary-button/PrimaryButtonWide.tsx
Comment (jzhzhang66) [inline shared-ui/components/primary-button/PrimaryButtonWide.tsx]: This PrimaryButton seems pretty identical to the original one so instead of creating this button, could we pass in a prop to the original button that's like "isSmall" or something and then add the extra padding based on that? 

Like this where we conditionally set the design based on the prop - this would get rid of a lot of code I think!
Screenshot 2023-01-05 at 6 22 30 PM

File: shared-ui/components/secondary-button/SecondaryButtonWhiteText.tsx
Comment (jzhzhang66) [inline shared-ui/components/secondary-button/SecondaryButtonWhiteText.tsx]: Same here - lets remove this file and pass in a prop so we can use the same button to handle both styles

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.styles.tsx
Comment (jzhzhang66) [inline live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.styles.tsx]: delete padding-left and right - it's centered perfectly without those styles and actually when we have them it causes it to not be centered on mobile LOL because you already have the justify-content: center

Screenshot 2023-01-05 at 6 32 44 PM

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx
Comment (jzhzhang66) [inline live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: Like Mike said, this links back to the home page - we should conditionally set the btnLink based on the schedulePersonType

File: live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx
Comment (jzhzhang66) [inline live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx]: change to padding: 10em 0; 

Its already centered so we don't need all the extra padding on the left and right since it causes it to be more smushed on mobile

Just be sure to stack the buttons on mobile. Also, to please address the button style inconsistencies that @mike10911 pointed out
Screenshot 2023-01-05 at 6 36 50 PM

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.styles.tsx
Comment (jzhzhang66) [inline live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.styles.tsx]: Could we please remove all the random new lines in the styled components?

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Thank you for all your hard work I know this ticket was a bit crazy LOL

This is looking great!! Left a few comments lmk if u have any questions!

- On mobile, make the header wider and also stack the buttons (you can do this by removing the left and right padding - I left another more specific comment for this)  
Screenshot 2023-01-05 at 6 29 58 PM

**Comment by jzhzhang66** [comment]
@mike10911 we decided to leave the arrow like this for now since it's actually quite difficult to change it from the standard one LOL - we can circle back to fixing that when the site is almost complete!! 

+1 to everything else you mentioned though!!

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx
Comment (surbhi-gulati) [reply live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: works correctly for me? we'd like it to go back to judging selection page regardless of personType which this handles appropriately right now!

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx
Comment (jzhzhang66) [inline live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: This still goes to the home page - it should go to the who are you page since that's the previous page right?

File: shared-ui/components/primary-button/PrimaryButton.styles.ts
Comment (jzhzhang66) [inline shared-ui/components/primary-button/PrimaryButton.styles.ts]: I think there is something going wrong with the logic somewhere because the previous regular primary buttons are now super skinny 
Screenshot 2023-01-09 at 10 57 29 PM

File: shared-ui/components/primary-button/PrimaryButton.tsx
Comment (jzhzhang66) [inline shared-ui/components/primary-button/PrimaryButton.tsx:30]: I wonder if the issue I mentioned has to do with this being called isSmallPrimary but you're passing in isSmall as your prop

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Nice job getting the props to work!! 

I think there's a bit of regression on the basic PrimaryButton so I left some comments about that and also that the back button redirects to the home page instead of the who are you page

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Screenshot 2023-01-09 at 11 02 34 PM
These buttons need to be stacked on mobile

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Border of button should be green not white
Screenshot 2023-01-09 at 11 04 10 PM
Screenshot 2023-01-09 at 11 04 17 PM

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx
Comment (surbhi-gulati) [reply live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: @mike10911 we are going to let this go because for some reason it works well in deployment, just not localhost

File: live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx
Comment (jzhzhang66) [inline live-site/components/judging-section/judging-schedule-welcome/JudgingScheduleWelcome.tsx]: remove these {' '}

File: live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx
Comment (jzhzhang66) [inline live-site/components/judging-section/judging-selection/JudgingSelection.styles.tsx:27]: Could you please replace lines 27 - 42 with the following? Currently, you're centering them by adding a bunch of left styling but we can reduce this code a lot if we just apply styling to center them! 

```
const StyledJudgeButton = styled.div`
  @media ${max.tablet} {
    margin-bottom: 2em;
    justify-content: center;
    display: flex;
  }
`;

const StyledHackerButton = styled.div`
  margin-left: 2em;
  @media ${max.tablet} {
    margin-left: 0;
    justify-content: center;
    display: flex;
  }
`;

```

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
THIS LOOKS GREAT!!!! its almost there! I just left two small requests. The button stacking css logic can be simplified a lot so I suggested a change for that! And there's also some random spaces so that's the other change i requested.

ILLL BE READY TO APPROVE AFTER!!

**Comment by surbhi-gulati** [comment]
@mike10911 hope you're feeling better today!! can you give a re-re-review when you get the chance?

**Review by jzhzhang66 — APPROVED** [review]
LGTM THANKS SURBHI!

**Review by mike10911 — APPROVED** [review]
NO CRUMBS LEFT. LGTM

**Pull request merged**
